src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -18,7 +18,7 @@ public abstract partial class Workspace
         private readonly Dictionary<ProjectId, ISet<DocumentId>> _projectToOpenDocumentsMap = new Dictionary<ProjectId, ISet<DocumentId>>();
 
         // text buffer maps
-        private readonly Dictionary<SourceTextContainer, DocumentId> _bufferToDocumentInCurrentContextMap = new Dictionary<SourceTextContainer, DocumentId>();
+        private readonly Dictionary<SourceTextContainer, OneOrMany<DocumentId>> _bufferToDocumentInCurrentContextMap = new Dictionary<SourceTextContainer, OneOrMany<DocumentId>>();
```

Comment:
It's a little strange for the "_bufferToDocumentInCurrentContextMap" to map to multiple documents, given its name and purpose, and the fact that the 0th element has to be _special_ in a non-obvious way. I was imagining some secondary data structure to track all of the documentIds related to the SourceTextContainer and then removing the SourceTextContainer from the dictionary when all of the documentIds are removed for it.

I won't block merging based on this, but do take a minute to consider this alternate approach. I'd like to know what you think about it.